src/runtime_src/xdp/profile/plugin/aie_profile_new/edge/aie_profile.h

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright (C) 2022 Advanced Micro Devices, Inc. - All rights reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef AIE_PROFILE_H
+#define AIE_PROFILE_H
+
+#include <cstdint>
+
+#include "core/edge/common/aie_parser.h"
+#include "xdp/profile/plugin/aie_profile_new/aie_profile_impl.h"
+#include "xaiefal/xaiefal.hpp"
+
+extern "C" {
+#include <xaiengine.h>
+#include <xaiengine/xaiegbl_params.h>
+}
+
+namespace xdp {
+  using tile_type = xrt_core::edge::aie::tile_type;
+  
+  class AieProfile_EdgeImpl : public AieProfileImpl{
+    public:
+      // AieProfile_EdgeImpl(VPDatabase* database, std::shared_ptr<AieProfileMetadata> metadata)
+      //   : AieProfileImpl(database, metadata){}
+      AieProfile_EdgeImpl(VPDatabase* database, std::shared_ptr<AieProfileMetadata> metadata);
+
+      ~AieProfile_EdgeImpl() = default;
+
+      void updateDevice();
+      void poll(uint32_t index, void* handle);
+      bool checkAieDevice(uint64_t deviceId, void* handle);
+
+      bool setMetricsSettings(uint64_t deviceId, void* handle);
+      std::vector<tile_type> getAllTilesForCoreMemoryProfiling(const XAie_ModuleType mod,
```

Comment:
These functions don't make sense here as x86 will also need this metadata. As we discussed in meeting, xAie_ModuleType isn't needed here and this can be refactored to be at a common place. Please makes these changes when implementing x86 side of things.